src/components/table/sortable/index.js

```diff
@@ -0,0 +1,89 @@
+import React, { Component } from 'react';
+import {
+  arrayOf,
+  objectOf,
+  oneOfType,
+  string,
+  number,
+  shape,
+} from 'prop-types';
+import classNames from 'classnames';
+
+import { TableRows } from '../helpers';
+
+const TableSortable = class TableSortable extends Component {
+  static propTypes = {
+    headers: arrayOf(
+      shape({
+        title: string.isRequired,
+        key: string.isRequired,
+      }),
+    ).isRequired,
+    rows: arrayOf(objectOf(oneOfType([string, number]))).isRequired,
+    classnames: string,
+  };
+  static defaultProps = {
+    classnames: '',
+  };
+  constructor(props) {
+    super(props);
+    const { rows, headers } = props;
+    this.state = {
+      rows,
+      headers: headers.map(header => ({ ...header, sortDirection: 'asc' })),
+    };
+  }
+  componentWillReceiveProps(nextProps) {
+    const { rows, headers } = nextProps;
+    this.setState({
+      rows,
+      headers: headers.map(header => ({
+        sortDirection: this.state.headers.filter(
+          stateHeader => stateHeader.key === header.header,
+        )[0].sortDirection,
+        ...header,
+      })),
+    });
+  }
+  sort = (key, direction) => {
```

Comment:
I would kinda expect to be able to swap out this behaviour with a HTTP based sorting. Any thoughts?

Reply:
Yep. I added support for bypassing this functionality if you provide your own callback.